// janohirmas-printgametable/index.js
module.exports = GameTable;

function GameTable(content,iR,iC,vColNames,vRowNames) {
    this.Table = content; 
    let length = content.length;
    // Rows
    if (!iR && !iC) {
        let sqrt= Math.sqrt(length) ;
        if (Number.isInteger(sqrt)) {
            iC = sqrt;
            iR = sqrt;
        }
    }
    if (iR) {
        if (length%iR == 0) {
            if (!iC) {
                iC = length/iR;
            };
        } else {
            console.log('Rows do not fit in table');
        };
    };
    // Columns
    if (iC) {
        if (length%iC == 0) {
            if (!iR) {
                iR = length/iC;
            };
        } else {
            console.log('Columns do not fit in table');
        };
    };
    // Check both match 
    if (length/(iR*iC)!=1) {
        console.log('Dimensions do not match');
    }
    this.Rows = iR;
    this.Columns = iC;
    const ABC = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
    
    if (vColNames) {
        this.ColNames = vColNames;
    } else {
        this.ColNames = ABC.slice(-iC);
    }
    if (vRowNames) {
        this.RowNames = vRowNames;
    } else {
        this.RowNames = ABC.slice(0,iR);
    }
}
